monkey/infection_monkey/i_puppet/target_host.py

```diff
@@ -15,6 +15,14 @@ class TargetHostPorts(MutableInfectionMonkeyBaseModel):
     tcp_ports: Dict[NetworkPort, PortScanData] = Field(default={})
     udp_ports: Dict[NetworkPort, PortScanData] = Field(default={})
 
+    @property
+    def closed_tcp_ports(self) -> Set[NetworkPort]:
+        return {
+            port
+            for port, port_scan_data in self.tcp_ports.items()
+            if port_scan_data.status == PortStatus.CLOSED
+        }
```

Comment:
This was added and used in the last two commits. I could take or leave it. I'll let democracy decide which way was better.

Reply:
I think it is good

Reply:
Having slept on this, I'm going to take one more pass at it.